RCTOpenTokPublisherView.m

```diff
@@ -240,12 +244,21 @@ - (void)session:(OTSession*)session didFailWithError:(OTError*)error {
 
 #pragma mark - OTSession delegate - archive callbacks
 - (void)session:(OTSession *)session archiveStartedWithId:(NSString *)archiveId name:(NSString *)name {
+    if (_recordTimer == nil) {
+        NSMutableArray *params = [NSMutableArray array];
+        [params addObject:archiveId];
+        [params addObject:name];
+        _recordTimer = [NSTimer scheduledTimerWithTimeInterval:1.5 target:self selector:@selector(_progressTimerFired:) userInfo:params repeats:NO];
+    }
+}
+
+- (void)_progressTimerFired:(NSTimer *)progressTimer {
     [self saveThumbnail];
```

Comment:
probably is more accurate to `saveThumnail` after calling `_onArchiveStarted` (after notifying the user)